flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -613,25 +614,31 @@ protected void cleanUpInvoke() throws Exception {
 		// stop all timers and threads
 		tryShutdownTimerService();
 
+		Throwable suppressedThrowable = null;
 		// stop all asynchronous checkpoint threads
 		try {
 			cancelables.close();
 			shutdownAsyncThreads();
 		} catch (Throwable t) {
-			// catch and log the exception to not replace the original exception
-			LOG.error("Could not shut down async checkpoint threads", t);
+			// catch and suppress the exception to not replace the original exception
+			suppressedThrowable = ExceptionUtils.firstOrSuppressed(t, suppressedThrowable);
```

Comment:
It would be nice to preserve `"Could not shut down async checkpoint threads"` message (by wrapping `t`); but probably stacktrace is enough.

Reply:
What is the suggestion here?

Reply:
I suggest 
> wrapping `t`

```
suppressedThrowable = ExceptionUtils.firstOrSuppressed(
    new RuntimeException("Could not shut down async checkpoint threads", t),
    suppressedThrowable);
```